test/intl402/DateTimeFormat/prototype/resolvedOptions/hourCycle-dateStyle.js

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Mozilla Corporation, Igalia S.L. All rights reserved.
+// This code is governed by the license found in the LICENSE file.
+
+/*---
+esid: sec-Intl.DateTimeFormat.prototype.resolvedOptions
+description: >
+  Intl.DateTimeFormat.prototype.resolvedOptions properly
+  reflect hourCycle settings when using dateStyle.
+features: [Intl.DateTimeFormat-datetimestyle]
+---*/
+
+const hcValues = ["h11", "h12", "h23", "h24"];
+const hour12Values = ["h11", "h12"];
+
+for (const dateStyle of ["full", "long", "medium", "short"]) {
+  assert.sameValue(new Intl.DateTimeFormat([], { dateStyle }).resolvedOptions().dateStyle,
+                   dateStyle,
+                   `Should support dateStyle=${dateStyle}`);
+
+  /* Values passed via unicode extension key work */
+
+  for (const hcValue of hcValues) {
+    const resolvedOptions = new Intl.DateTimeFormat(`de-u-hc-${hcValue}`, {
+      dateStyle,
+    }).resolvedOptions();
+
+    assert.sameValue(resolvedOptions.hourCycle, undefined);
```

Comment:
Why is this undefined? 
https://tc39.es/proposal-intl-datetime-style/#sec-initializedatetimeformat
```
21. Set dateTimeFormat.[[HourCycle]] to hc.
```